witness/src/Dscp/Witness/Web/Logic.hs

```diff
@@ -246,9 +247,18 @@ checkFairCVPDF
     => Pdf.PDFBody -> m FairCVAndCheckResult
 checkFairCVPDF pdf = do
     let maybeFairCV = do
-            fairCVencoded <- Pdf.project (Pdf.MaxSearchLength Nothing) pdf
-            Aeson.decodeStrict fairCVencoded
+            (fairCVencoded, Pdf.PDFBody source) <- Pdf.unInject (Pdf.MaxSearchLength Nothing) pdf
+            res <- Aeson.decodeStrict fairCVencoded
+            return (res, source)
+
+    (fairCV, source) <- maybe (throwM InvalidFormat) pure maybeFairCV
+    checkRes         <- checkFairCV fairCV
+
+    let pdfHashIsValid =
+            (  fairCV^?to(fcCV).each.each.to(toList).each.ptSignedSubmission.ssSubmission.sContentsHash
```

Comment:
You still compare the PDF hash with contents hash of _every_ transaction, while you actually need only first one. In the certificate, there is only one Educator and only one proof per Educator, so I believe that changing the last `each` to `_head` would do.

Reply:
Really? The `^?` operator returns `Maybe a` - which is the first one, if any is present.